src/Platform/Vulkan/VulkanContext.cc

```diff
@@ -506,4 +511,20 @@ static void create_logical_device(ContextData& context_data, DeviceContextData&
 
   vkGetDeviceQueue(device_context_data.m_device, indices.m_graphics_family, 0, &context_data.m_graphics_queue);
   vkGetDeviceQueue(device_context_data.m_device, indices.m_present_family, 0, &context_data.m_present_queue);
+}
+
+static VkSampleCountFlagBits get_max_usable_sample_count()
+{
+  VkPhysicalDeviceProperties properties = esp::VulkanDevice::get_properties();
+
+  VkSampleCountFlags counts =
+      properties.limits.framebufferColorSampleCounts & properties.limits.framebufferDepthSampleCounts;
+  if (counts & VK_SAMPLE_COUNT_64_BIT) { return VK_SAMPLE_COUNT_64_BIT; }
+  if (counts & VK_SAMPLE_COUNT_32_BIT) { return VK_SAMPLE_COUNT_32_BIT; }
+  if (counts & VK_SAMPLE_COUNT_16_BIT) { return VK_SAMPLE_COUNT_16_BIT; }
+  if (counts & VK_SAMPLE_COUNT_8_BIT) { return VK_SAMPLE_COUNT_8_BIT; }
+  if (counts & VK_SAMPLE_COUNT_4_BIT) { return VK_SAMPLE_COUNT_4_BIT; }
+  if (counts & VK_SAMPLE_COUNT_2_BIT) { return VK_SAMPLE_COUNT_2_BIT; }
+
+  return VK_SAMPLE_COUNT_1_BIT;
```

Comment:
Suggestion: adding ```if(!counts) return VK_SAMPLE_COUNT_1_BIT;``` before all other if's will allow you to drop the "counts &&" part of other conditions.

Reply:
Bruh "&" and "&&" ain't same things you should know that:D